flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/execution/MailboxProcessor.java

```diff
@@ -286,12 +288,20 @@ public void resume() {
 			if (taskMailboxExecutor.isMailboxThread()) {
 				resumeInternal();
 			} else {
-				try {
-					mailbox.putMail(this::resumeInternal);
-				} catch (InterruptedException ie) {
+				boolean delivered = false;
+				boolean interrupted = false;
+				while (!delivered) {
```

Comment:
With the latest update to my PR, making the mailbox unbounded, this whole fix should no longer be required.